### PR TITLE
MSBuild for pipeline

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by janos-ss** [comment]
LGTM, but will it pass Travis when merged?

**Comment by henryju** [comment]
I think it is a FP, but I have restarted the job on travis to be sure.

File: src/main/java/hudson/plugins/sonar/MsBuildSQRunnerBegin.java [hunk 1]
```diff
@@ -67,13 +69,20 @@
...
 
   @DataBoundConstructor
+  public MsBuildSQRunnerBegin() {
+  }
+
+  /**
+   * @deprecated since 2.5. Moved to use {@link DataBoundSetter}
+   */
+  @Deprecated
   public MsBuildSQRunnerBegin(String msBuildScannerInstallationName, String sonarInstallationName, String projectKey, String projectName, String projectVersion,
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/MsBuildSQRunnerBegin.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/hudson/plugins/sonar/MsBuildSQRunnerBegin.java [hunk 2]
```diff
@@ -67,13 +69,20 @@
...
-  private final String projectVersion;
-  private final String additionalArguments;
+  private String sonarInstallationName;
+  private String projectKey;
+  private String projectName;
+  private String projectVersion;
+  private String additionalArguments;
 
   @DataBoundConstructor
+  public MsBuildSQRunnerBegin() {
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/MsBuildSQRunnerBegin.java]: CRITICAL Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1186)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* CRITICAL 1 critical
* INFO 1 info

Watch the comments in this conversation to review them.

**Pull request merged**